eZ/Bundle/EzPublishCoreBundle/Command/CleanupVersionsCommand.php

```diff
@@ -119,7 +136,13 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         $status = $input->getOption('status');
 
-        $contentIds = $this->getObjectsIds($keep, $status);
+        $contentTypeIds = explode(',', $input->getOption('excluded-content-types'));
+        $excludedContentTypeIds = array_unique(array_merge(
```

Comment:
I'm a bit unsure about always forcing this on the options, as opposed to let user be in control over ´excluded-content-types´ argument. The default is fine tough, it can be a good practice to skip users by default for misc reasons.

Why?
`ContentService::publishVersion()` _(also used by UserService)_ also removes drafts if you have reached version limits. So we really should try to figure out where in the kernel the "issues on deleting user draft" issue happens so we can properly solve it for all uses of version removal.

WDYT @alongosz and @adamwojs ?

Reply:
> I'm a bit unsure about always forcing this on the options, as opposed to let user be in control over `excluded-content-types` argument. The default is fine tough, it can be a good practice to skip users by default for misc reasons.

For me it makes sense to skip user by default if not given anything. User is unfortunately a special case.

> `ContentService::publishVersion()` _(also used by UserService)_ also removes drafts if you have reached version limits. So we really should try to figure out where in the kernel the "issues on deleting user draft" issue happens so we can properly solve it for all uses of version removal.

This is true, but the script is probably needed anyway. Even if we find the root cause, we still need to cleanup pre-existing erroneous data.

Reply:
> This is true, but the script is probably needed anyway. Even if we find the root cause, we still need to cleanup pre-existing erroneous data.

You kind of missing my point.
- Yes this script is still needed _(upgrades for instance, or to flatten version database)_, AND the improvement here is good 👍 
- No, forcing `user` as identifier without any way to overload that on this command, does *not* solve the underlying unconfirmed issue with user and version removal. The issue, if even true, will still most likely be present when updating user objects _(triggering publish, which triggers versions removal, ... same code path here afaics)_  🐛.

Reply:
Based on your arguments, I decided to get rid of forcing `user` CT to be excluded by default. I have no proof that the reported issues have been caused by removing versions belonging to users, just an unconfirmed suspicion. It requires further investigation.